airflow/models.py

```diff
@@ -920,6 +920,17 @@ def is_queueable(
         else:
             return False
 
+
+    def is_premature(self):
```

Comment:
Very close to being redundant with `not ready_for_retry()` the difference is very subtle.